Generalize PaddedSpectrumWCS for higher dimensionality

rosteen opened the pull request:
## Description

This is an alternative to #61 that rudely ignores @astrofrog's comment explicitly saying not to do this 😆 . The upshot is that it was far more clear to me how to do this than how to finish out the other PR, and we do need to fix #60  as soon as possible. 

Note that I've also taken the opportunity to stop reordering the axes of the data at all here, leaving it to the user to handle  `Spectrum1D` order with the spectral axis last (as opposed to `SpectralCube`, which has that axis first). That seems far better than swapping axes for 3D data but not 2D (or 4D?) data for reasons that were really downstream and shouldn't have been on `glue-astronomy` to deal with anyway. This means that, if this does end up getting merged, it will require some fixes to be ready in `jdaviz` (I'm working on these now).

astrofrog commented on glue_astronomy/translators/spectrum1d.py:
```diff
@@ -42,24 +42,25 @@ class PaddedSpectrumWCS(BaseWCSWrapper, HighLevelWCSMixin):
     # generalize this we can just remove this class and use CompoundLowLevelWCS
```
Update comment above

astrofrog replied:
Update comment above

astrofrog reviewed (changes requested):
This looks great, thanks! Small comment below, and also wanted to check if you could add some simple tests for the methods/properties that codecov is complaining about in the diff, just as we might as well take this opportunity to make sure they work fine (as glue relies on some of them).

Also thank you thank you thank you for no longer re-ordering axes :bow: :laughing: 

Once this is merged in, I guess we should fix jdaviz to be able to use this version, then I can tag a release of glue-astronomy. We should maybe talk at the SME tag-up next week about how to coordinate releases of glue-astronomy and jdaviz to make sure we don't impact users.

rosteen commented:
I fixed a few remaining issues with the world<->pixel conversion and added test coverage for the 3D case, I think this is ready for final review now. I still have a decent bit of work on the related Jdaviz PR.

astrofrog commented:
@rosteen - it looks like this needs a rebase because of the recent changelog changes.

astrofrog commented:
Thanks!
